### PR TITLE
Add concourse and FAKE build artifacts

### PR DESCRIPTION
**Pull request by lclarksl**

### CODE REVIEW

**Comment by jebbinSCL** [comment]
:shipit:

File: concourse/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM mono
+
+COPY binaries/ service/
+
+EXPOSE 9004
```
Comment (csscottlogic) [inline concourse/Dockerfile]: This is fine, but I did quickly check that we hadn't already used this port number on a different image.
When we get the overlay network working, each container will look like a separate host on the overlay network, so all our microservices can listen on port 80, and we will be able to simply call ```http://comms/```

File: ApiGateway/packages.config [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-  <package id="Microsoft.AspNet.WebApi.WebHost" version="5.2.3" targetFramework="net452" />
-  <package id="Microsoft.Owin" version="2.0.2" targetFramework="net452" />
-  <package id="Microsoft.Owin.Host.HttpListener" version="2.0.2" targetFramework="net452" />
-  <package id="Microsoft.Owin.Hosting" version="2.0.2" targetFramework="net452" />
-  <package id="Newtonsoft.Json" version="6.0.4" targetFramework="net452" />
-  <package id="Owin" version="1.0" targetFramework="net452" />
-  <package id="Serilog" version="1.5.14" targetFramework="net452" />
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Bristech.Srm.HttpConfig" version="0.1.1.0" targetFramework="net45" />
```
Comment (csscottlogic) [inline ApiGateway/packages.config]: I think this has increased to 0.1.2.0, but if the API gateway works with this version, then that's fine. The idea with the nuget packages is that microservices can upgrade independently anyway.

File: build.fsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+let buildDir = "./build/output"
+
+Target "Clean" (fun _ ->
+    CleanDir buildDir
+)
+
+Target "BuildApp" (fun _ ->
+    !! "ApiGateway/**/*.fsproj"
+    |> MSBuildRelease buildDir "Build"
+    |> Log "AppBuld-Output: "
```
Comment (csscottlogic) [inline build.fsx]: Build

File: build/pipeline/build.sh [hunk 4]
```diff
@@ -0,0 +1 @@
+docker build -t bristechsrm/build_image .
```
Comment (csscottlogic) [inline build/pipeline/build.sh]: The definition for the build-fsharp image has moved to infrastructure. We no longer need build/pipeline/Dockerfile or build.sh here

File: concourse/task-images/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu
```
Comment (csscottlogic) [inline concourse/task-images/Dockerfile]: This Dockerfile, along with it's associated .sh script has moved to infrastructure

Comment (lclarksl) [inline concourse/task-images/Dockerfile]: Meaning the task-images folder is not needed?

**Comment by csscottlogic** [comment]
Apart from removing Dockerfile's and their associated shell scripts for building the F# compiler image (bristechsrm/build-fsharp), :shipit:

**Pull request merged**